src/passes/OptimizeInstructions.cpp

```diff
@@ -1117,6 +1117,18 @@ struct OptimizeInstructions
   }
 
   void visitRefEq(RefEq* curr) {
+    // Canonicalize nulls to make the next pattern simpler (and also it may
+    // compress better).
```

Comment:
Not sure what this means... Aren't we converting this into `ref.is_null` after all? Not sure "next pattern simpler" and "compress better" means.

Reply:
Good point, yeah, that wasn't very clear... I rewrote it. Also the compression part is actually inaccurate - if there is a null, we optimize it out anyhow, so we can't be helping compression... I removed that bit.